### PR TITLE
Remove linear regression parameters from harmonic model

### PR DESCRIPTION
**Pull request by veni-vidi-vici-dormivi**
We do not fit a linear regression of the seasonal cycle with the yearly predictor, so we can move the sum with the yearly predictor from the deepest level (`generate_fourier_series`) to the highest one (`fit_harmonic_model`). See also [this comment](https://github.com/MESMER-group/mesmer/pull/484#discussion_r1705753163).

### CODE REVIEW

File: mesmer/stats/_harmonic_model.py [hunk 1]
```diff
@@ -270,21 +265,25 @@ def fit_harmonic_model(yearly_predictor, monthly_target, max_order=6, time_dim="
         yearly_predictor, monthly_target[time_dim], time_dim=time_dim
     )
 
+    seasonal_deviations = monthly_target - yearly_predictor
```
Comment (mathause) [inline mesmer/stats/_harmonic_model.py:268]:
```suggestion
    # subtract annual mean to have seasonal anomalies around 0
    seasonal_deviations = monthly_target - yearly_predictor
```

File: tests/unit/test_harmonic_model.py [hunk 2]
```diff
@@ -18,28 +19,26 @@ def test_generate_fourier_series_np():
     n_years = 10
     n_months = n_years * 12
 
-    yearly_predictor = np.zeros(n_months)
+    yearly_predictor = np.ones(n_months)
     months = np.tile(np.arange(1, 13), n_years)
 
+    coeffs = np.array([0, -1, 0, -2])
```
Comment (mathause) [inline tests/unit/test_harmonic_model.py:25]: nice

**Review by mathause — APPROVED** [review]
Thanks for your extra work. Should we add changelog entries for the three PRs?

**Comment by veni-vidi-vici-dormivi** [comment]
Okay I added entries to the changelog, also two things that don't correspond to this PR, is it still fine if I merge it here and not do another PR?

File: CHANGELOG.rst [hunk 3]
```diff
@@ -100,17 +105,21 @@ that this led to some numerical changes compared to the MESMER-M publication
...
    - only fit orders up to local minimum and use coeffs from precious order as first guess (`#443 <https://github.com/MESMER-group/mesmer/pull/443>`_)
    - infer the harmonic model order from the coefficients (`#434 <https://github.com/MESMER-group/mesmer/pull/434>`_)
 -  return residuals instead of the loss for the optimization (`#460 <https://github.com/MESMER-group/mesmer/pull/460>`_)
+-  remove fitting of linear regression with yearly temperature (`#415 <https://github.com/MESMER-group/mesmer/pull/415/>_` and
+   `#488 <https://github.com/MESMER-group/mesmer/pull/488>`_) in line with (`Nath et al. 2022 <https://doi.org/10.5194/esd-13-851-2022>`_).
 -  add helper function to upsample yearly data to monthly resolution (
    `#418 <https://github.com/MESMER-group/mesmer/pull/418>`_, and
    `#435 <https://github.com/MESMER-group/mesmer/pull/435>`_)
 - de-duplicate the expression of months in their harmonic form (`#415 <https://github.com/MESMER-group/mesmer/pull/415>`_)
+  and move creation of the month array to the deepest level (`#487 <https://github.com/MESMER-group/mesmer/pull/487>`_).
```
Comment (mathause) [inline CHANGELOG.rst:122]:
```suggestion
  move creation of the month array to the deepest level (`#487 <https://github.com/MESMER-group/mesmer/pull/487>`_).
```

File: CHANGELOG.rst [hunk 4]
```diff
@@ -14,6 +14,9 @@ Breaking changes
 ^^^^^^^^^^^^^^^^
 - Using Cholesky decomposition for finding covariance localization radius and drawing from the multivariate normal distribution (`#408 <https://github.com/MESMER-group/mesmer/pull/408>`_)
   By `Victoria Bauer`_.
+- Ensure the power transformer yields the correct normalization for more cases see `Yeo-Johnson power transformer`_.
+- The auto-regression for monthly variables now has no bounds anymore and the covariance is estimated on the residuals
+  of the AR(1) instead of adjusting the empirical covariance matrix with the AR slopes see `Auto-Regression`_.
```
Comment (mathause) [inline CHANGELOG.rst:19]: I would mention all the MESMER stuff below - these comments are now duplicated. I don't find them more (or less) important than others and we'd have to think about what other things to highlight. So I'd remove them again.

**Review by mathause — COMMENTED** [review]
> Okay I added entries to the changelog, also two things that don't correspond to this PR, is it still fine if I merge it here and not do another PR?

Sure not everything needs to have its own PR

File: CHANGELOG.rst (hunk 4 above)
Comment (veni-vidi-vici-dormivi) [reply CHANGELOG.rst:19]: Okay

**Pull request merged**